c/tskit/haplotype_matching.c

```diff
@@ -570,18 +570,19 @@ all_zero(const uint64_t *restrict A, size_t u, size_t num_words)
 }
 
 static inline bool
-element_in(const uint64_t *restrict A, size_t u, const tsk_id_t state, size_t num_words)
+element_in(
+    const uint64_t *restrict A, tsk_size_t u, const tsk_id_t state, tsk_size_t num_words)
 {
-    size_t index = ((size_t) u) * num_words + (size_t)(state / 64);
+    tsk_size_t index = ((tsk_size_t) u) * num_words + (tsk_size_t)(state / 64);
```

Comment:
Actually, I think it's the parameter u is the wrong type - it should be tsk_id_t. So, the cast is OK I think, since we're casting tsk_id_t to tsk_size_t.